e2e/solve/test.go

```diff
@@ -384,71 +401,81 @@ func openOrder(ctx context.Context, backends ethbackend.Backends, order TestOrde
 	}, solvernet.WithFillDeadline(order.FillDeadline))
 }
 
-func testCheckAPI(ctx context.Context, orders []TestOrder) error {
+func testCheckAPI(ctx context.Context, backends ethbackend.Backends, orders []TestOrder) error {
 	const url = "http://localhost:26661/api/v1/check"
 
-	var eg errgroup.Group
 	for i, order := range orders {
-		eg.Go(func() error {
-			checkReq := solver.CheckRequest{
-				FillDeadline:       uint32(order.FillDeadline.Unix()), //nolint:gosec // this is fine for tests
-				SourceChainID:      order.SourceChainID,
-				DestinationChainID: order.DestChainID,
-				Expenses:           expensesFromBindings(order.Expenses),
-				Calls:              callsFromBindings(order.Calls),
-				Deposit:            addrAmtFromDeposit(order.Deposit),
+		// If this order requires balance draining, do it before test logic.
+		if isInsufficientInventory(order) {
+			// Drain solver native balance.
+			if err := setSolverAccountNativeBalance(ctx, backends, big.NewInt(0)); err != nil {
```

Comment:
I would also pass in the chain on which the balance should be set, in this case `DestinationChainID` (not hard coded like now)